tests/dead_test.py

```diff
@@ -18,6 +18,36 @@ def test_dead_disable_regex_not_matching(s):
     assert not dead.DISABLE_COMMENT_RE.search(s)
 
 
+@pytest.mark.parametrize('suffix', ('', ' # dead: disable'))
+@pytest.mark.parametrize(
+    'string, matches',
+    (
+        ('# type: (int, int) -> float', True),
+        ('# type: ignore', True),
+        ('# type: ClassVar[str]', True),
+    ),
+)
+def test_type_regex_matching(string, matches, suffix):
+    assert bool(dead.TYPE_COMMENT_RE.search(string + suffix)) is matches
+
+
+@pytest.mark.parametrize('suffix', ('', ' # dead: disable'))
+@pytest.mark.parametrize(
+    'string, matches',
+    (
+        ('#type:ignore', True),
+        ('# type: ignore', True),
+        ('# type: ignore[no-any-return]', True),
+        ('# type: ignore   [no-any-return]', True),
+        ('# type: ignore[$omething]', False),
+        ('# type: ignore{no-any-return}', False),
+        ('# type: ignore[]', False),
```

Comment:
I'd rather see positive and negative tests separated instead of boolean-trap parametrize